### PR TITLE
Update JXL to v0.1.

### PR DESCRIPTION
**Pull request by veluca93**

### CODE REVIEW

File: codecs/jxl/Makefile [hunk 1]
```diff
@@ -51,18 +51,8 @@ $(CODEC_OUT): $(CODEC_DIR)/CMakeLists.txt
...
-	# seems you can’t clone a specific commit directly), and it also means
-	# we can’t use --depth 1 because we want to change commits.
-	# The JXL code base also relies on submodules so we can’t just download
-	# a .tar.gz from GitLab.
 	mkdir -p $@
-	git clone $(CODEC_URL) $@
-	cd $@ && \
-	git reset --hard $(CODEC_VERSION) && \
-	git submodule update --init --recursive
+	git clone $(CODEC_URL) --recursive --depth 1 --branch $(CODEC_VERSION) $@
```
Comment (RReverser) [inline codecs/jxl/Makefile]: Ha, on another branch I just recently made a different change to allow parallel submodules even without `clone`, but this simplifies it a lot. Just one suggestion - let's parallelise it:

```suggestion
	git clone $(CODEC_URL) --recursive -j`nproc` --depth 1 --branch $(CODEC_VERSION) $@
```

File: codecs/jxl/enc/jxl_enc.cpp [hunk 2]
```diff
@@ -64,7 +64,7 @@ val encode(std::string image, int width, int height, JXLOptions options) {
     cparams.responsive = 0;
   }
 
-  io.metadata.SetAlphaBits(8);
+  io.metadata.m.SetAlphaBits(8);
```
Comment (RReverser) [inline codecs/jxl/enc/jxl_enc.cpp:67]: Hmm, what is `m.`? It's not very self-descriptive :)

**Comment by jakearchibald** [comment]
@veluca93 can you remove the `package-lock.json` change from this PR?

File: codecs/jxl/enc/jxl_enc.cpp (hunk 2 above)
Comment (veluca93) [reply codecs/jxl/enc/jxl_enc.cpp:67]: `metadata.m` is what used to be `metadata` - this is a bit of an intermediate state between refactors...

**Comment by veluca93** [comment]
> @veluca93 can you remove the `package-lock.json` change from this PR?

Done. I imagine it got modified by doing something with npm...

**Comment by jakearchibald** [comment]
I've unmerged this. It built locally fine, but it couldn't decode the JXLs it created.

**Pull request merged**